src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,14 +34,17 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        ShoppingCart cart = shoppingCartService
+                .getByUser(userService.findByEmail(authentication.getName())
+                .orElseThrow(() -> new RuntimeException("Invalid email")));
```

Comment:
message -> Can't find user by email: authentication.getName()
the same for other similar cases